### DISCUSSION TITLE
Introducing Prompts Extras

### DISCUSSION THREAD

**Discussion by henzeb** [laravel/framework#50241] [Show and tell]
Prompts Extras, a package  that offers additional functionalities that expand upon Laravel's Prompt. While these features may not be universally desired, they could prove valuable for specific use cases. 

https://packagist.org/packages/henzeb/prompts-extras

Right now, it only has `watch`, which works similar to `watch` found on Linux/Unix systems. 

```php
use function Henzeb\Prompts\watch;
use function Laravel\Prompts\table;

watch(
    render: function (){
        table(
            ['current date'],
            [
                [(new Datetime())->format(DateTime::RFC850)]
            ]
        );
    },
    interval: 1 // default interval is 2
);
```

If you have a feature idea, that doesn't fit into Laravel Prompts, but still think it might be useful to some, let's add that one here.

**Comment by henzeb** [comment]
Just wanted to announce an update: With `Prompts Extras`, output with Prompts is isolated within commands, and validation is done using Laravel's own validation mechanism.

In addition, `Prompts Extras` allows validation to be done automatically on all your options and arguments. I know Laravel has something built in, but I believe this is way nicer and better grouped together. And it allows the same behavior in non-laravel projects.

**Comment by shyamsitaula** [comment]
@henzeb I have used Laravel prompts in my project https://github.com/oitcode/samarium . Hopefully someday can use this.
